### PR TITLE
Examples Framework & Fractal example

### LINKED ISSUE #30: Add Examples

### PR DESCRIPTION
**Pull request by 9prady9**
Fixes #30

### CODE REVIEW

File: examples/CMakeModules/FindGLEWmx.cmake [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+OPTION(USE_GLEWmx_STATIC "Use Static GLEWmx Lib" OFF)
+
+FIND_PATH(GLEW_INCLUDE_DIR GL/glew.h
+    HINTS
+    ${GLEW_ROOT_DIR}/include
+    )
+
+IF (WIN32)
+	IF (USE_GLEWmx_STATIC)
```
Comment (shehzan10) [inline examples/CMakeModules/FindGLEWmx.cmake]: Can you changes these lines to use spaces instead of block tabs?

Comment (shehzan10) [inline examples/CMakeModules/FindGLEWmx.cmake]: There are other files which use block tab as well. Please fix those too.

File: examples/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    IF(${FORGE_FOUND})
+        BUILD_EXAMPLE(${EXAMPLE} ${FILE} ${DIR_NAME} ${FORGE_LIBRARIES} "")
+    ELSEIF(TARGET forge)
+        BUILD_EXAMPLE(${EXAMPLE} ${FILE} ${DIR_NAME} forge "")
+    ENDIF()
+ENDFOREACH()
+
+FIND_PACKAGE(CUDA)
+IF(CUDA_FOUND)
+    FIND_PACKAGE(NVVM REQUIRED)
```
Comment (shehzan10) [inline examples/CMakeLists.txt]: NVVM is not required for what we are doing in the examples. It can be completely removed.

**Comment by shehzan10** [comment]
Any chance we could get away with not including cl.hpp as a part of the repo?

File: examples/CMakeLists.txt (hunk 2 above)
Comment (9prady9) [inline examples/CMakeLists.txt]: Don't see if any other example might need it ?

**Comment by 9prady9** [comment]
Fixed the tabs. Regarding `cl.hpp` any reason ? examples will be easy to write, otherwise we have to write examples with horrible C API and it is more error prone.

File: examples/CMakeLists.txt (hunk 2 above)
Comment (shehzan10) [inline examples/CMakeLists.txt]: No. NVVM is for JIT/IR and other stuff. None of which we will be doing in these examples.

**Comment by shehzan10** [comment]
I meant use the system cl.hpp. I'm not sure whether it's guaranteed to be on every machine, that's why I asked if there is a possibility of not having it here.

**Comment by 9prady9** [comment]
It wont be there on every machine, especially for CUDA toolkits. NVIDIA doesn't provide the `cl.hpp`. AMD APPSDK does i think. It is not uniform yet, so if we want to use it we have to provide it.

**Comment by shehzan10** [comment]
Ok.

**Comment by 9prady9** [comment]
Removed NVVM dependency.

**Comment by shehzan10** [comment]
Sorry for pestering, but the tabs are still there in CMakeModules/FindGLEWmx.cmake. I think you only cleared the examples one.
That said, why do we need 2 of the same files?

**Comment by 9prady9** [comment]
i did a retab in vim, i wonder how come they are still there. When examples are shipped with installers, wouldn't that need FindGLEW for that ?

**Comment by 9prady9** [comment]
oh wait, you said the main FindGLEWmx . ok i didnt touch that file. might have been there from earlier itself.

**Comment by 9prady9** [comment]
Fixed tabs in that FindGLEWmx file.

**Comment by 9prady9** [comment]
I have sent that white space fix too.

**Comment by 9prady9** [comment]
@shehzan10 It is ready for merge now.

**Pull request merged**